astropy/time/core.py

```diff
@@ -282,6 +283,8 @@ def _get_time_fmt(self, val, val2, format, scale):
             try:
                 return FormatClass(val, val2, scale, self.precision,
                                    self.in_subfmt, self.out_subfmt)
+            except UnitConversionError:
```

Comment:
OK, yes, I think I now understand: that much be the interaction with `Quantity` in `TimeDelta`